### ISSUE TITLE
strpos deprecated warning

### ISSUE THREAD

**Issue by tcarroll2** [Admidio/admidio#884]
I am getting this warning in my PHP (v7.3.1) log file:

```
[12-Jul-2019 11:05:10 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 75
[12-Jul-2019 11:05:10 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 76
```

**Comment by ximex** [comment]
what do you get in `ADMIDIO_URL_PATH`?
https://github.com/Admidio/admidio/blob/master/adm_program/system/bootstrap/constants.php#L66

**Comment by tcarroll2** [comment]
Looking at the variable in my debugger It is null.

**Comment by tcarroll2** [comment]
This is all that is in my Admidio log:

`[2019-07-12 13:40:01.075363] Admidio.ALERT: DATABASE: Could not connect to Database! EXCEPTION MSG: SQLSTATE[HY000] [2002] No such file or directory {"engine":"mysql","host":"localhost","port":8889,"dbName":"admidio-v4-beta","username":"admidio","password":"******","options":[]} {"file":"/Users/Shared/admidio-4_0-beta/adm_program/system/classes/Database.php","line":217,"class":"Database","function":"connect"}
`
Which I think this was when mySQL server wasn't running.

**Comment by tcarroll2** [comment]
The variable is empty.

**Comment by ximex** [comment]
do you have set the `$g_root_path`?

**Comment by tcarroll2** [comment]
It is set to 'http://admidio-4-beta:8888’

**Comment by tcarroll2** [comment]
I am developing on a Mac using MAMP Pro with the latest version for Admidio, MacOS, MAMP Pro and Chrome.

**Comment by ximex** [comment]
change it to `http://admidio-4-beta:8888/` and it should work ;-)
but we should check that better

**Comment by tcarroll2** [comment]
I added the backslash to the end of the URL in the config file as recommending.  It did not fix the issue.  This is what I am getting in my php.log file:

`[13-Jul-2019 12:57:07 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 75
[13-Jul-2019 12:57:07 America/Chicago] PHP Stack trace:
[13-Jul-2019 12:57:07 America/Chicago] PHP   1. {main}() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:0
[13-Jul-2019 12:57:07 America/Chicago] PHP   2. require_once() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:24
[13-Jul-2019 12:57:07 America/Chicago] PHP   3. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/common.php:28
[13-Jul-2019 12:57:07 America/Chicago] PHP   4. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/bootstrap.php:21
[13-Jul-2019 12:57:07 America/Chicago] PHP   5. strpos(*uninitialized*, *uninitialized*) /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php:75
[13-Jul-2019 12:57:07 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 76
[13-Jul-2019 12:57:07 America/Chicago] PHP Stack trace:
[13-Jul-2019 12:57:07 America/Chicago] PHP   1. {main}() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:0
[13-Jul-2019 12:57:07 America/Chicago] PHP   2. require_once() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:24
[13-Jul-2019 12:57:07 America/Chicago] PHP   3. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/common.php:28
[13-Jul-2019 12:57:07 America/Chicago] PHP   4. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/bootstrap.php:21
[13-Jul-2019 12:57:07 America/Chicago] PHP   5. strpos(*uninitialized*, *uninitialized*) /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php:76`

**Comment by tcarroll2** [comment]
I added the backslash to the end of the URL in the config file as recommending.  It did not fix the issue.  This is what I am getting in my php.log file:

`[13-Jul-2019 12:57:07 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 75
[13-Jul-2019 12:57:07 America/Chicago] PHP Stack trace:
[13-Jul-2019 12:57:07 America/Chicago] PHP   1. {main}() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:0
[13-Jul-2019 12:57:07 America/Chicago] PHP   2. require_once() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:24
[13-Jul-2019 12:57:07 America/Chicago] PHP   3. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/common.php:28
[13-Jul-2019 12:57:07 America/Chicago] PHP   4. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/bootstrap.php:21
[13-Jul-2019 12:57:07 America/Chicago] PHP   5. strpos(*uninitialized*, *uninitialized*) /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php:75
[13-Jul-2019 12:57:07 America/Chicago] PHP Deprecated:  strpos(): Non-string needles will be interpreted as strings in the future. Use an explicit chr() call to preserve the current behavior in /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php on line 76
[13-Jul-2019 12:57:07 America/Chicago] PHP Stack trace:
[13-Jul-2019 12:57:07 America/Chicago] PHP   1. {main}() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:0
[13-Jul-2019 12:57:07 America/Chicago] PHP   2. require_once() /Users/Shared/admidio-4_0-beta/adm_program/modules/photos/photo_show.php:24
[13-Jul-2019 12:57:07 America/Chicago] PHP   3. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/common.php:28
[13-Jul-2019 12:57:07 America/Chicago] PHP   4. require_once() /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/bootstrap.php:21
[13-Jul-2019 12:57:07 America/Chicago] PHP   5. strpos(*uninitialized*, *uninitialized*) /Users/Shared/admidio-4_0-beta/adm_program/system/bootstrap/constants.php:76`

**Comment by tcarroll2** [comment]
This fixed it.  Thank you!

**Comment by Fasse** [comment]
This was fix with #923
